lyse/__init__.py

```diff
@@ -43,29 +38,18 @@
 
 # Import this way so LYSE_DIR is exposed when someone does import lyse or from lyse import *
 from lyse.utils import LYSE_DIR
-
-# If running stand-alone, and not from within lyse, the below two variables
-# will be as follows. Otherwise lyse will override them with spinning_top =
-# True and path <name of hdf5 file being analysed>:
-spinning_top = False
-# data to be sent back to the lyse GUI if running within lyse
-_updated_data = {}
-# dictionary of plot id's to classes to use for Plot object
-_plot_classes = {}
-# A fake Plot object to subclass if we are not running in the GUI
-Plot=object
-# An empty dictionary of plots (overwritten by the analysis worker if running within lyse)
-plots = {}
-# A threading.Event to delay the 
-delay_event = threading.Event()
-# a flag to determine whether we should wait for the delay event
-_delay_flag = False
-
-if len(sys.argv) > 1:
-    path = sys.argv[1]
-else:
-    path = None
-
+from lyse.utils.worker import spinning_top, _updated_data, register_plot_class, delay_results_return
+
+# note: path is injected into the script namespace by AnalysisWorker at runtime
+def __getattr__(name):
+    if name == 'path':
+        from lyse.utils.worker import path
+        warnings.warn("'path' is now automatically injected into the script namespace. "
+                      "Importing it from 'lyse.path' will be deprecated.",
+                      warnings.FutureWarning)
```

Comment:
This warning definition is wrong. It should just be `FutureWarning`

Reply:
I would just fix it, but it appears I can't push commits to your branch anymore (probably for good reason).